src/lib/accounts/index.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  AccountBase,
+  sign,
+  messageToHash,
+  RpcRejectedByUserError,
+  unpackTx,
+} from '@aeternity/aepp-sdk-13';
+import { Encoded } from '@aeternity/aepp-sdk-13/src/utils/encoder';
+import { Store } from 'vuex';
+import { useAccounts } from '../../composables/accounts';
+import { IS_CORDOVA, IS_EXTENSION_BACKGROUND } from '../environment';
+import { POPUP_TYPE_MESSAGE_SIGN, POPUP_TYPE_SIGN, POPUP_TYPE_TX_SIGN } from '../../popup/utils';
+import { showPopup } from '../../background/popupHandler';
+
+export class AccountSuperhero extends AccountBase {
+  address: Encoded.AccountAddress;
+
+  store: Store<any>;
+
+  aeppInfo: Record<string, any> = {};
+
+  constructor(store: Store<any>) {
+    super();
+    this.store = store;
+    const { account } = useAccounts({ store: this.store });
+    this.address = account.value.address as Encoded.AccountAddress;
```

Comment:
Maybe it would be better to store whole `account` in the scope of the class to allow using it instead of `store.getters.account` ? This would help to migrate from Vuex.

Reply:
since the account is used for only one function at the `sign` message. I am using the accounts composables at the sign method itself and not at the class level.

Reply:
OK